msbuild/Xamarin.MacDev.Tasks/Tasks/LinkNativeCodeTaskBase.cs

```diff
@@ -196,8 +196,13 @@ bool ExecuteUnsafe ()
 			arguments.Add (Path.GetFullPath (OutputFile));
 
 			if (LinkerFlags is not null) {
-				foreach (var flag in LinkerFlags)
-					arguments.Add (flag.ItemSpec);
+				foreach (var flag in LinkerFlags) {
+					// If a LinkerFlags item includes a whitespace it will be passed along to clang surrounded by quotes.
+					// This can cause unexpected behavior with command line parsing and failures.
+					// To prevent this, split the string and pass its parts as individual arguments instead. 
+					var flagParts = flag.ItemSpec.Split (new char [0], StringSplitOptions.RemoveEmptyEntries);
+					arguments.AddRange (flagParts);
+				}
```

Comment:
See above, if you add each argument separately to the item group, this won't be necessary.

Reply:
Fixed.